### PR TITLE
fix(providers): handle cop_marine in situ historical data

### PR DESCRIPTION
**Pull request by jlahovnik**
`cop_marine` provider fixes:
- if no date is given in the id of the product, the date given for the dataset the product belongs to is now used
- a mapping has been added to retrieve other information from the id (such as platform, analysis type) and to add it to the properties of the product

### CODE REVIEW

**Review by dalpasso — APPROVED** [review]

File: eodag/plugins/search/cop_marine.py [hunk 1]
```diff
@@ -348,16 +372,54 @@ def query(
...
+                                item_end = datetime.strptime(
+                                    item_end, "%Y-%m-%dT%H:%M:%S.%f%z"
+                                )
+                            except ValueError:
+                                item_start = datetime.strptime(
+                                    item_start, "%Y-%m-%dT%H:%M:%S%z"
+                                )
+                                item_end = datetime.strptime(
+                                    item_end, "%Y-%m-%dT%H:%M:%S%z"
+                                )
```
Comment (sbrunato) [inline eodag/plugins/search/cop_marine.py:410]: Argument 1 to "strptime" of "datetime" has incompatible type "datetime | None"; expected "str"mypy[arg-type](https://mypy.readthedocs.io/en/stable/_refs.html#code-arg-type)

File: eodag/plugins/search/cop_marine.py [hunk 2]
```diff
@@ -348,16 +372,54 @@ def query(
...
+                    if item_dates:
+                        item_start = _get_date_from_yyyymmdd(item_dates[0], item_key)
+                        if len(item_dates) > 2:  # start, end and created_at timestamps
+                            item_end = _get_date_from_yyyymmdd(item_dates[1], item_key)
+                    if not item_start:
+                        # no valid datetime given in id
+                        use_dataset_dates = True
+                        dates = _get_dates_from_dataset_data(dataset_item)
+                        if dates:
+                            item_start = dates["start"].replace("Z", "+0000")
```
Comment (sbrunato) [inline eodag/plugins/search/cop_marine.py:395]: Incompatible types in assignment (expression has type "str", variable has type "datetime | None")mypy[assignment](https://mypy.readthedocs.io/en/stable/_refs.html#code-assignment)

File: eodag/plugins/search/cop_marine.py [hunk 3]
```diff
@@ -348,16 +372,54 @@ def query(
...
+                            item_end = dates["end"].replace("Z", "+0000")
```
Comment (sbrunato) [inline eodag/plugins/search/cop_marine.py:396]: Incompatible types in assignment (expression has type "str", variable has type "datetime | None")mypy[assignment](https://mypy.readthedocs.io/en/stable/_refs.html#code-assignment)

**Review by sbrunato — DISMISSED** [review]

**Comment by jlahovnik** [comment]
type problems fixed

**Review by sbrunato — APPROVED** [review]

**Pull request merged**